collector/lib/rust/host/src/info.rs

```diff
@@ -0,0 +1,224 @@
+use crate::KernelVersion;
+
+use std::fs::File;
+use std::io::{prelude::*, BufReader};
+use std::path::PathBuf;
+
+const MIN_RHEL_BUILD_ID: u64 = 957;
+
+// The values are taken from efi_secureboot_mode in include/linux/efi.h
+#[derive(Default)]
+pub enum SecureBootStatus {
+    #[allow(dead_code)]
+    Enabled = 3,
+    #[allow(dead_code)]
+    Disabled = 2,
+
+    // Secure Boot seems to be disabled, but the boot loaded
+    // does not provide enough information about it,
+    // so it could be enabled without the kernel being aware.
+    #[allow(dead_code)]
+    NotDetermined = 1,
+
+    // No detection is performed yet
+    #[default]
+    Unset = 0,
+}
+
+pub struct HostInfo {
+    os: String,
+    build: String,
+    distro: String,
+    hostname: String,
+    kernel: KernelVersion,
+}
+
+impl HostInfo {
+    pub fn new() -> Self {
+        let mut os = String::new();
+        let mut build = String::new();
+        let mut distro = String::new();
+        let mut hostname = String::new();
+
+        if let Ok(file) = File::open("/etc/os-release") {
+            let reader = BufReader::new(file);
+
+            for line in reader.lines() {
+                match line {
+                    Ok(s) if s.starts_with("ID") => os = s["ID ".len()..].to_owned(),
+                    Ok(s) if s.starts_with("BUILD_ID") => build = s["BUILD_ID ".len()..].to_owned(),
+                    Ok(s) if s.starts_with("PRETTY_NAME") => {
+                        distro = s["PRETTY_NAME ".len()..].to_owned()
+                    }
+                    Ok(_) => {}
+                    Err(_) => break,
+                };
+            }
+        }
+
+        let hostname_paths = vec![
+            host_path("/etc/hostname"),
+            host_path("/proc/sys/kernel/hostname"),
+        ];
+        for path in hostname_paths {
+            if let Ok(mut file) = File::open(path) {
+                if file.read_to_string(&mut hostname).is_ok() {
+                    break;
+                }
+            }
+        }
+
+        HostInfo {
+            os,
+            build,
+            distro,
+            hostname,
+            kernel: KernelVersion::from_host(),
+        }
+    }
+
+    /// Whether or not this host machine is COS
+    pub fn is_cos(&self) -> bool {
+        self.os_id().eq("cos") && !self.build_id().is_empty()
+    }
+
+    /// Whether or not this host machine is CORE-OS
+    pub fn is_coreos(&self) -> bool {
+        self.os_id().eq("coreos")
```

Comment:
Could we use standard `==` in these checks?
```suggestion
        self.os_id() == "coreos"
```